superset-frontend/src/SqlLab/components/TemplateParamsEditor.jsx

```diff
@@ -68,23 +60,26 @@ export default class TemplateParamsEditor extends React.Component {
     } catch (e) {
       isValid = false;
     }
-    this.setState({ parsedJSON, isValid, codeText });
+    setState({ parsedJSON, isValid, codeText });
     const newValue = isValid ? codeText : '{}';
-    if (newValue !== this.props.code) {
-      this.props.onChange(newValue);
+    if (newValue !== code) {
+      onChange(newValue);
     }
-  }
+  };
+
+  useEffect(() => {
+    onChange(codeText);
+  }, [codeText]);
```

Comment:
don't think you need this useEffect either. The component will automatically update state if the codeText is changed in either setState or from the props.